lab-geoff/server.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+let http = require('http');
+let url = require('url');
+let querystring = require('querystring');
+let cowsay = require('cowsay');
+let reqHandler = require('./lib/requestHandler.js');
+
+let PORT = process.env.PORT || 3000;
+
+let server = http.createServer(function(req, res) {
+  req.url = url.parse(req.url);
+  console.log(req.url.path);
+  req.url.query = querystring.parse(req.url.query);
+  console.log(req.url.query);
+
+  if(req.method === 'POST') {
+    console.log('post request block');
+    reqHandler(req, function(string) {
+      if (string.text) {
+        res.writeHead(200, {
+          'Content-type': 'text/plain',
+        });
+        res.write(cowsay.say({
+          text: `\n${string.text}\n`,
+        }) + '\n');
+        res.end();
+      } else {
+        console.log('no text in body');
+        res.writeHead(400, {
+          'Content-type': 'text/plain',
+        });
+        res.write(cowsay.say({
+          text: 'bad request\ntry: localhost:3000/cowsay?text=howdy',
+        }) + '\n');
+        res.end();
+      }
+    });
+  }
+
+  else if(req.method === 'GET' && req.url.pathname === '/cowsay') {
+    console.log('cowsay get request block');
```

Comment:
Once you get a piece of code working, remove some or all of the console.logs related to the piece. Keep your code clean, so you know what your logs are for when they fire off.